scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -586,4 +587,26 @@ object TreeOps {
   def hasExplicitImplicit(param: Term.Param): Boolean =
     param.mods.exists(isExplicitImplicit)
 
+  def shouldNotDangleAtDefnSite(
+      tree: Tree,
+      isVerticalMultiline: Boolean
+  )(implicit style: ScalafmtConfig): Boolean =
+    !style.danglingParentheses.defnSite || {
+      val excludeList =
+        if (isVerticalMultiline && style.danglingParentheses.exclude.isEmpty)
+          style.verticalMultiline.excludeDanglingParens
+        else
+          style.danglingParentheses.exclude
+      excludeList.nonEmpty && {
+        val exclude = tree match {
+          case _: Ctor.Primary | _: Defn.Class =>
+            DanglingParentheses.Exclude.`class`
+          case _: Defn.Trait => DanglingParentheses.Exclude.`trait`
+          case _: Defn.Def => DanglingParentheses.Exclude.`def`
+          case _ => null
+        }
+        null != exclude && excludeList.contains(exclude)
```

Comment:
`excludeList.contains(null) == false`

Reply:
if you don't mind, I'd leave it as-is. the intent is more explicit and doesn't require a list scan for something definitely not there

Reply:
👌